Description: Append -latomic on RISCV, as some atomics need libatomic support.
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/cmake/+bug/1876814
Author: Dimitri John Ledkov <xnox@ubuntu.com>


--- a/Modules/Platform/Linux.cmake
+++ b/Modules/Platform/Linux.cmake
@@ -57,3 +57,15 @@
   set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB32_PATHS FALSE)
   set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS FALSE)
 endif()
+
+# RISCV needs -latomic
+# https://gitlab.kitware.com/cmake/cmake/-/issues/20895
+#
+# Ideally I'd use CMAKE_C_COMPILER_ARCHITECTURE_ID, but that is only
+# available on Win platforms. Not sure why though, cause every
+# compiler on Linux has a default architecture it compiles for too,
+# when no flags are passed to ask the compiler to behave differently.
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv64")
+  string(APPEND CMAKE_C_STANDARD_LIBRARIES " -latomic")
+  string(APPEND CMAKE_CXX_STANDARD_LIBRARIES " -latomic")
+endif()
